finalfusion-ffi/src/vocab.rs

```diff
@@ -0,0 +1,73 @@
+use std::ffi::CStr;
+use std::mem;
+use std::os::raw::{c_char, c_int};
+
+use finalfusion::chunks::vocab::WordIndex;
+use finalfusion::prelude::*;
+
+use crate::check_null;
+
+/// Indices lookup.
+///
+/// Returns an integer indicating how a word is represented:
+///    * -1 if the word cannot be represented
+///    * 0 if the word is in-vocabulary
+///    * `n` if the word is represented by `n` embeddings
+///
+/// Writes the indices to `indices_res` if it is not a null-pointer.
+#[no_mangle]
+pub unsafe extern "C" fn ff_vocab_indices(
```

Comment:
I see why you would want to distinguish word and subword embeddings. However, this interface has two downsides:

1. In C you could not simply do a `if (ff_vocab_indices(...))` to check whether indices could be found.
2. You cannot use the return value as the length of `indices_res`. You need to handle the in-vocab word separately.

I think I would be more in favor of this method returning `0` (the word cannot be represented) vs. `n` (the length). An providing a separate function that simply returns `0/1` to check if a word is in-vocab.

Reply:
I can live with that, do you first want to merge #21? I'll amend this PR with the additional method

Reply:
Sorry, didn't see you had approved that. Should maybe first finish my coffee.

Reply:
> I can live with that, do you first want to merge #21? I'll amend this PR with the additional method

The method now returns `0` for unrepresentable words and `n` for all representables. I'll open a follow-up PR with a `contains` method later, but might wait for a `finalfusion-rust` release.